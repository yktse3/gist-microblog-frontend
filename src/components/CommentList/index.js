import React from 'react';
import { string, shape, arrayOf } from 'prop-types';
import Comment from 'components/Comment';
import {
  CommentBlock,
} from './styles';

const CommentList = ({ comments }) => (
  <CommentBlock>
    {
      comments.map(comment => (
        <Comment
          userID={comment.userID}
          content={comment.content}
          key={comment.id}
        />
      ))
    }
  </CommentBlock>
);

CommentList.propTypes = {
  comments: arrayOf(
    shape({
      id: string,
      userID: string,
      content: string,
    }),
  ).isRequired,
};

export default CommentList;
