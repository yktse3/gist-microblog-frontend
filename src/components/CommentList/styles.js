import styled from 'styled-components';

export const CommentBlock = styled.div`
  border-top: 1px solid #888481;
`;
