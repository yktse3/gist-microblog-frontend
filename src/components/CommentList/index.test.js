import React from 'react';
import renderer from 'react-test-renderer';
import CommentList from './index';

describe('CommentList', () => {
  it('render successfully', () => {
    const component = renderer.create(
      <CommentList
        comments={
          [
            {
              userID: 'username',
              content: 'comment content',
              id: 'commentID',
            },
            {
              userID: 'username2',
              content: 'comment content2',
              id: 'commentID2',
            },
          ]
        }
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
