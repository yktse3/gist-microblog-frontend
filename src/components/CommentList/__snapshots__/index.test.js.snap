// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList render successfully 1`] = `
<div
  className="sc-EHOje dtHEGr"
>
  <div
    className="sc-htpNat eCCHOJ"
  >
    <span
      className="sc-bdVaJa hQekPs"
    >
      username replied:
    </span>
    <span
      className="sc-bwzfXH jmosDp"
    >
      comment content
    </span>
  </div>
  <div
    className="sc-htpNat eCCHOJ"
  >
    <span
      className="sc-bdVaJa hQekPs"
    >
      username2 replied:
    </span>
    <span
      className="sc-bwzfXH jmosDp"
    >
      comment content2
    </span>
  </div>
</div>
`;
